crates/wit-parser/src/resolve.rs

```diff
@@ -1861,6 +1915,7 @@ impl Remap {
         &mut self,
         resolve: &mut Resolve,
         iface: &mut Interface,
+        iface_pkg_id: &PackageId,
```

Comment:
Could this be inferred from `iface.pkg.unwrap()` instead of being passed in perhaps?

Reply:
So I could have sworn here that I ran into an issue where the `iface` that came in *did not* have the right package on it... But now that I look back, they always seem to. Don't know what happened here.

Ah, so after rechecking, I run into this *inside* the `append()` code, at precisely here:

https://github.com/bytecodealliance/wasm-tools/blob/c811041f3549dd93a889bd99ebb80ef33757152c/crates/wit-parser/src/resolve.rs#L1549

So on the very next line `iface.package` is *supposed* to be `None`, but it turns out it's *also* `None` on the line *before* that -- I assumed this was because this fn was called while going through`UnresolvedPackage`s.

It *seems* save to simply update the package first (as the code does a few lines down), so I changed it to do that instead -- please let me know if that violates some assumption that `update_interface` was making previously (I can't see one).

https://github.com/bytecodealliance/wasm-tools/pull/1689/commits/cb4dbf9b3577600a97eef2c75171ec544e32f280